pkg/envoy/proxy.go

```diff
@@ -29,6 +31,8 @@ type Proxy struct {
 	lastAppliedVersion map[TypeURI]uint64
 	lastNonce          map[TypeURI]string
 
+	lastxDSResourcesSent map[TypeURI]mapset.Set
```

Comment:
It is tough to know what is in the set.  I wonder if a more concrete type instead of a set would be better?
I wonder if sync.Map (in this PR or eventually) would make this a) saver and b) clear what this is